src/aere_sophia.erl

```diff
@@ -28,12 +53,12 @@ typecheck(Ast, Opts) ->
     end.
 
 compile_contract(TypedAst) ->
-    {#{child_con_env := ChildConEnv
-     } = FCodeEnv, FCode} = try aeso_ast_to_fcode:ast_to_fcode(TypedAst, [])
-                 catch {error, Ec} -> process_err(Ec) end,
-    SavedFreshNames = maps:get(saved_fresh_names, FCodeEnv, #{}),
-    try aeso_fcode_to_fate:compile(ChildConEnv, FCode, SavedFreshNames, [include_child_contract_symbols]) of
-        {Code, _} -> Code
+    Opts = [debug_info, include_child_contract_symbols | optimizations_off()],
+    {#{child_con_env := ChildConEnv, saved_fresh_names := SavedFreshNames}, FCode}
+        = try aeso_ast_to_fcode:ast_to_fcode(TypedAst, Opts)
+          catch {error, Ec} -> process_err(Ec) end,
+    try
+        aeso_fcode_to_fate:compile(ChildConEnv, FCode, SavedFreshNames, Opts)
     catch {error, Ef} -> process_err(Ef) end.
 
 type_of_user_input(TEnv) ->
```

Comment:
Crash:
```
AESO> 2
Command failed:
{badmatch,#{}}
Stacktrace:
[{aeso_ast_infer_types,ets_tabid,1,
....
```
I am using OTP 25, but I don't think it's an issue. It looks like a bug in the compiler actually. Do all tests pass there?

Reply:
Same for `"type`. Interestingly, it displays type errors correctly.

Reply:
Did I fix this here 0f1716f52973622f4001eb628c967814a44c365e? ;)